src/node/coins_view_args.h

```diff
@@ -5,11 +5,20 @@
 #ifndef BITCOIN_NODE_COINS_VIEW_ARGS_H
 #define BITCOIN_NODE_COINS_VIEW_ARGS_H
 
+#include <kernel/caches.h>
+
 class ArgsManager;
 struct CoinsViewOptions;
 
-namespace node {
-void ReadCoinsViewArgs(const ArgsManager& args, CoinsViewOptions& options);
+namespace node
+{
+    static constexpr size_t GetDbBatchSize(const size_t dbcache_bytes)
+    {
+        const auto target{(dbcache_bytes / DEFAULT_KERNEL_CACHE) * DEFAULT_DB_CACHE_BATCH};
```

Comment:
In commit "coins: introduce dynamic batch size calculator based on dbcache value" (b249239c1747a5425e5161974d2b9c9c5be17dab)

Because division is happening before multiplication here, target will be 0 for any value below DEFAULT_KERNEL_CACHE, instead of scaling smoothly. This works well enough for current code because MIN_DB_CACHE_BATCH and DEFAULT_DB_CACHE_BATCH are equal, but if the minimum was less, this would return a discontinuous result instead of scaling smoothly. Would suggest either adding a conditional to handle this case or just using the default as the minimum explicitly here.